wiiu/main.c

```diff
@@ -163,35 +166,58 @@ void MCPHookClose(void)
    mcp_hook_fd = -1;
 }
 
-static void fsdev_init(void)
+static bool try_init_iosuhax(void)
 {
-   iosuhaxMount = 0;
-   int res = IOSUHAX_Open(NULL);
+   int result = IOSUHAX_Open(NULL);
+   if(result < 0)
+      result = MCPHookOpen();
 
-   if (res < 0)
-      res = MCPHookOpen();
+   return (result < 0) ? false : true;
+}
 
-   if (res < 0)
-      mount_sd_fat("sd");
-   else
-   {
-      iosuhaxMount = 1;
+static void try_shutdown_iosuhax(void)
+{
+  if(!iosuhaxMount)
+    return;
+
+  if (mcp_hook_fd >= 0)
+    MCPHookClose();
+  else
+    IOSUHAX_Close();
+
+  iosuhaxMount = false;
+}
+
+__attribute__((weak))
+void mount_filesystems(void)
+{
+   if(iosuhaxMount)
       fatInitDefault();
-   }
+   else
+      mount_sd_fat("sd");
 }
 
-static void fsdev_exit(void)
+__attribute__((weak))
+void unmount_filesystems(void)
```

Comment:
maybe `__mount_filesystems` and `__unmount_filesystems` instead ? just to reduce the chance of them being overridden by mistake, since they will not throw link errors when that happens.